accord-core/src/main/java/accord/local/Status.java

```diff
@@ -131,8 +290,45 @@ public boolean hasBeen(Status equalOrGreaterThan)
         return compareTo(equalOrGreaterThan) >= 0;
     }
 
-    public boolean isAtLeast(Phase equalOrGreaterThan)
+    public static <T> T max(List<T> list, Function<T, Status> getStatus, Function<T, Ballot> getAccepted, Predicate<T> filter)
+    {
+        T max = null;
+        Status maxStatus = null;
+        Ballot maxAccepted = null;
+        for (T item : list)
+        {
+            if (!filter.test(item))
+                continue;
+
+            Status status = getStatus.apply(item);
+            Ballot accepted = getAccepted.apply(item);
+            boolean update = max == null
+                          || maxStatus.phase.compareTo(status.phase) < 0
+                          || (status.phase.equals(Phase.Accept) && maxAccepted.compareTo(accepted) < 0);
+
+            if (!update)
+                continue;
+
+            max = item;
+            maxStatus = status;
+            maxAccepted = accepted;
+        }
+
+        return max;
+    }
+
+    public static <T> T max(T a, Status statusA, Ballot acceptedA, T b, Status statusB, Ballot acceptedB)
+    {
+        int c = statusA.phase.compareTo(statusB.phase);
+        if (c > 0) return a;
+        if (c < 0) return b;
+        if (statusA.phase != Phase.Accept || acceptedA.compareTo(acceptedB) >= 0)
```

Comment:
For my own understanding In phases after accept is it supposed to be an an invariant that the ballot doesn't change anymore?

I started asking questions about special casing because I was naively expecting to just blindly compare on ballot when phase is !=.

Reply:
Well technically no, you could still seek promises and accepts for newer ballots, the invariant is that they would be logically equivalent as the node doing that would always undertake the same action as any node with an earlier ballot that entered those states.